adapters/httpapi_compact.c

```diff
@@ -832,7 +832,7 @@ static HTTPAPI_RESULT SendHeadsToXIO(HTTP_HANDLE_DATA* http_instance, HTTPAPI_RE
     /*Codes_SRS_HTTPAPI_COMPACT_21_038: [ The HTTPAPI_ExecuteRequest shall execute the resquest for the path in relativePath parameter. ]*/
     /*Codes_SRS_HTTPAPI_COMPACT_21_036: [ The request type shall be provided in the parameter requestType. ]*/
     if (((ret = snprintf(buf, sizeof(buf), "%s %s HTTP/1.1\r\n", get_request_type(requestType), relativePath)) < 0) ||
-        (ret >= sizeof(buf)))
+        ((size_t)ret >= sizeof(buf)))
```

Comment:
>ret [](start = 17, length = 3)

Sounds like this should be a size_t?

Reply:
[https://linux.die.net/man/3/snprintf](https://linux.die.net/man/3/snprintf) returns an int and a negative number indicates an error. Not sure if there is anything else we can do here other than "up" type-casting 'ret' after checking that it is non negative.